### PR TITLE
Adding a units.yaml in config folder

### LINKED ISSUE #89: Standardize observation params
**What**: A configuration file that defines the name, the default units and other metadata for observation and impact metric params.
**Why**: So we can standardize language and expectations across all imps and the consumers of the framework.
**SoW**:
- When you see the word `cpu` in an observation list, it should be clear that this is for CPU utilization, the default units is percentage and to aggregate you average two utilizations, for things like carbon when you aggregate you sum them up.
- [ ] Create an IEF configuration file easy to find and refer to by all consumers of IEF that standardized these values.
- [ ] Make all existing IMPs and code adhere to this standard.

Example:

```yaml
cpu:
  description: refers to CPU utilization.
  unit: percentage
  aggregation: avg
carbon:
  description: an amount of carbon emitted into the atmosphere
  unit: gCO2e
  aggregation: sum
```

### PR DESCRIPTION
**Pull request by gnanakeethan**
### Types of changes

- [x] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### A description of the changes proposed in the Pull Request

Adding a units.yaml that specifies the units that will be used by the project and its components when interacting with one another. 

FIxes #89

### CODE REVIEW

**Review by narekhovhannisyan — APPROVED** [review]

File: src/config/units.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  description: amount of energy utilised by the component
+  unit: KWh
+  aggregation: sum
+embodied_emission:
+  description: Embodied Emissions of the component
+  unit: gCO2e
+  aggregation: sum
+expected_life_span:
+  description: Total Expected Lifespan of the Component in Years
+  unit: Years
```
Comment (jmcook1186) [inline src/config/units.yaml:35]: elsewhere we've been converting years -> seconds here (e.g. passing 126144000 to represent 4 years)

Comment (gnanakeethan) [reply src/config/units.yaml:35]: I think it is much easier to comprehend for a human when writing a YAML file to write in just years.

**Comment by narekhovhannisyan** [comment]
if unit values described in yaml file will be used in the codebase, we are adding complexity to the code (parsing yaml).
It can be represented as f.ex. json file to be readable for all kind of users.

**Review by jawache — CHANGES_REQUESTED** [review]
Looks good to me after applying comments below.

File: src/config/units.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+cpu:
+  description: refers to CPU utilization.
+  unit: percentage
+  aggregation: avg
+memory:
+  description: refers to GB of memory utilized.
+  unit: GB
```
[the remark this answers is not in the document]
Comment (gnanakeethan) [reply src/config/units.yaml:7]: @jawache the issue can be like of which the percentage is this value. eg: 30% of which capacity?

File: src/config/units.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  description: refers to GB of data written/read from disk
+  unit: GB
+  aggregation: sum
+duration:
+  description: refers to the duration of the observation
+  unit: s
+  aggregation: sum
+timestamp:
+  description: refers to the time of occurrence of the observation
+  unit: ISO8601 / RFC3339 compatible timestamp
```
[the remark this answers is not in the document]
Comment (gnanakeethan) [reply src/config/units.yaml:19]: https://stackoverflow.com/a/65221179 

We should better go with either of them but not likely both. RFC3339 is more human readable and it allows some flex.

File: src/config/units.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  aggregation: sum
+energy:
+  description: amount of energy utilised by the component
+  unit: KWh
+  aggregation: sum
+embodied_emission:
+  description: Embodied Emissions of the component
+  unit: gCO2e
+  aggregation: sum
+expected_life_span:
```
[the remark this answers is not in the document]
Comment (gnanakeethan) [reply src/config/units.yaml:33]: it is better not to keep hyphens?

File: src/config/units.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jawache) [reply src/config/units.yaml:7]: Agreed, but we can have two ram-util and ram-used which is clearer than just having memory.

File: src/config/units.yaml (hunk 1 above)
Comment (jawache) [reply src/config/units.yaml:35]: @gnanakeethan I've seen it also refered to as "expected-lifespan" in other places now so let's use "expected-lifespan" with a unit of "seconds"

File: src/config/units.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jawache) [reply src/config/units.yaml:19]: Sure that works for me

File: src/config/units.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (jawache) [reply src/config/units.yaml:33]: I prefer hyphens and the GSF as a whole is using hyphens over underscore.

**Review by jawache — CHANGES_REQUESTED** [review]

**Comment by jawache** [comment]
> if unit values described in yaml file will be used in the codebase, we are adding complexity to the code (parsing yaml).
> It can be represented as f.ex. json file to be readable for all kind of users.
> @jawache

@narekhovhannisyan after some more thought, given that models can be written in any language (in the case of shell models which are communicated with through a sub-process). I think it makes more sense for the units file to be in a common format, yaml. rather than JS which only JS apps can read.

File: src/config/units.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (gnanakeethan) [reply src/config/units.yaml:11]: adding, "ram-util", "ram-alloc" which can be more descriptive themselves.

**Comment by jawache** [comment]
@gnanakeethan I cleaned up a little with case and also after some thinking it makes more sense to call them

embodied-carbon and operational-carbon since emissions is a broader term and the values are just specifically carbon figures. You will need to change the models to reflect that! Sorry.

**Pull request merged**